homeassistant/components/airzone/sensor.py

```diff
@@ -85,42 +85,64 @@ async def async_setup_entry(
     """Add Airzone sensors from a config_entry."""
     coordinator = entry.runtime_data
 
-    sensors: list[AirzoneSensor] = [
-        AirzoneZoneSensor(
-            coordinator,
-            description,
-            entry,
-            system_zone_id,
-            zone_data,
-        )
-        for system_zone_id, zone_data in coordinator.data[AZD_ZONES].items()
-        for description in ZONE_SENSOR_TYPES
-        if description.key in zone_data
-    ]
-
-    if AZD_HOT_WATER in coordinator.data:
-        sensors.extend(
-            AirzoneHotWaterSensor(
-                coordinator,
-                description,
-                entry,
+    added_hotwater: bool = False
+    added_webserver: bool = False
+    added_zones: set[str] = set()
+
+    def _async_entity_listener() -> None:
+        """Handle additions of sensors."""
+
+        nonlocal added_hotwater
+        nonlocal added_webserver
+
+        entities: list[AirzoneSensor] = []
+
+        if not added_hotwater and AZD_HOT_WATER in coordinator.data:
+            entities.extend(
+                AirzoneHotWaterSensor(
+                    coordinator,
+                    description,
+                    entry,
+                )
+                for description in HOT_WATER_SENSOR_TYPES
+                if description.key in coordinator.data[AZD_HOT_WATER]
             )
-            for description in HOT_WATER_SENSOR_TYPES
-            if description.key in coordinator.data[AZD_HOT_WATER]
-        )
-
-    if AZD_WEBSERVER in coordinator.data:
-        sensors.extend(
-            AirzoneWebServerSensor(
-                coordinator,
-                description,
-                entry,
+            added_hotwater = True
+
+        if not added_webserver and AZD_WEBSERVER in coordinator.data:
```

Comment:
Can this be added on runtime? Or is this something that is there by default? 

You could also make separate listeners for these and if the webserver is already set up, we don't attach the listener

Reply:
Actually you're right, so on a second thought I've removed the webserver and water heater from the listeners since those can't be dynamically added.